static/js/ProfileDetail.tsx

```diff
@@ -0,0 +1,221 @@
+import React, { FC } from "react";
+import { useParams } from "react-router-dom";
+import { useQuery } from "@tanstack/react-query";
+import { fetchProfile } from "./api/profiles";
+import DeleteProfileBtn from "./buttons/profiles/DeleteProfileBtn";
+import EditProfileBtn from "./buttons/profiles/EditProfileBtn";
+import OpenProfileListBtn from "./buttons/profiles/OpenProfileListBtn";
+import BaseLayout from "./components/BaseLayout";
+import NotificationRow from "./components/NotificationRow";
+import { queryKeys } from "./util/queryKeys";
+import useNotification from "./util/useNotification";
+import {
+  CodeSnippet,
+  CodeSnippetBlockAppearance,
+  Col,
+  Row,
+} from "@canonical/react-components";
+import { LxdDiskDevice, LxdNicDevice } from "./types/device";
+
+const ProfileDetail: FC = () => {
+  const { name } = useParams<{ name: string }>();
+
+  if (!name) {
+    return <>Missing name</>;
+  }
+
+  const notify = useNotification();
+
+  const { data: profile, error } = useQuery({
+    queryKey: [queryKeys.profiles, name],
+    queryFn: () => fetchProfile(name),
+  });
+
+  if (error) {
+    notify.failure("Could not load profile details.", error);
+  }
+
+  if (!profile) {
+    return <>Could not load profile details.</>;
+  }
+
+  const btnProps = {
+    profile: profile,
+    notify: notify,
+    appearance: "",
+    className: "u-no-margin--bottom",
+    isDense: false,
+  };
+
+  const containsDevices = (type: "nic" | "disk") => {
```

Comment:
I think we could benefit from using the same approach as in this PR with isFooDevice helpers: https://github.com/canonical/lxd-ui/pull/120/files#diff-37a7120764c7cadda0eec0b5bf2b8783704700be479fb051f60a8d751382c69a

But that is not in Main yet....

Reply:
I added those helper functions and replaced `containsDevices` and `getDevices` with them @ [67c3f03](https://github.com/canonical/lxd-ui/commit/67c3f037beb960c5ca2c14b7ee095e108bda37de)